File: DotNET/src/Parameters.cs
// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
using System;
namespace src
{
    public class Parameters
    {
        public readonly string Input;
        public readonly string Output;
        public readonly bool Verbose;
        public Parameters(string[] args)
        {
            bool needAssignedIn = true;
            string inTemp = "input.txt";
            bool needAssignedOut = true;
            string outTemp = "output.txt";
            bool verboseTemp = false;
            for (int i = 0; i < args.Length; ++i)
            {
                if (needAssignedIn && "-in".Equals(args[i]))
                {
                    if (i < args.Length - 1)
                    {
                        inTemp = args[i + 1];
                        needAssignedIn = false;
                        ++i;
                    }
                }
                else if (needAssignedOut && "-out".Equals(args[i]))
                {
                    if (i < args.Length - 1)
                    {
                        outTemp = args[i + 1];
                        needAssignedOut = false;
                        ++i;
                    }
                }
                else if (!verboseTemp && "-verbose".Equals(args[i]))
                {
                    verboseTemp = true;
                }
            }
            Input = inTemp;
            Output = outTemp;
            Verbose = verboseTemp;
        }
    }
}

File: Typescript/src/parameters.ts
export class Parameters {
    public readonly in: string;
    public readonly out: string;
    public readonly verbose: boolean;

    public constructor(args: string[]) {
        let needAssignedIn = true;
        let inTemp = "input.txt";
        let needAssignedOut = true;
        let outTemp = "output.txt";
        let verboseTemp = false;
        for (let i = 0; i < args.length; i += 1) {
            if (needAssignedIn && args[i] === "-in") {
                if (i < args.length - 1) {
                    inTemp = args[i + 1];
                    needAssignedIn = false;
                    i += 1;
                }
            } else if (needAssignedOut && args[i] === "-out") {
                if (i < args.length - 1) {
                    outTemp = args[i + 1];
                    needAssignedOut = false;
                    i += 1;
                }
            } else if (!verboseTemp && args[i] === "-verbose") {
                verboseTemp = true;
            }
        }
        this.in = inTemp;
        this.out = outTemp;
        this.verbose = verboseTemp;
    }
}
